core/src/main/java/tc/oc/pgm/integrations/SimpleVanishIntegration.java

```diff
@@ -84,28 +84,12 @@ public List<MatchPlayer> getOnlineVanished() {
   public boolean setVanished(MatchPlayer player, boolean vanish, boolean quiet) {
     final Match match = player.getMatch();
 
-    // Call vanish event next so name renders existing state properly for leave msg
-    if (vanish) match.callEvent(new PlayerVanishEvent(player, vanish, quiet));
-
-    // Keep track of the UUID and apply/remove META data, so we can detect vanish status from other
-    // projects (i.e utils)
-    if (vanish) {
-      addVanished(player);
-    } else {
-      removeVanished(player);
-    }
-
-    // Ensure player is an observer
-    if (vanish && player.getParty() instanceof Competitor) {
-      match.setParty(player, match.getDefaultParty());
-    }
+    // Call vanish event which will vanish/unvanish at appropriate levels
+    match.callEvent(new PlayerVanishEvent(player, vanish, quiet));
```

Comment:
Thanks for taking care of the vanish fix! I was investigating it earlier in the day, excellent solution having the vanish logic extracted out of this method 👍